mono/metadata/unity-liveness.c

```diff
@@ -237,6 +276,59 @@ static gboolean mono_add_process_object(MonoObject *object, LivenessState *state
 	return FALSE;
 }
 
+MONO_API void mono_validate_object_pointer (MonoObject *object)
+{
+	if (object) {
+		void *vtable = NULL;
+		MonoClass *klass = NULL;
+		char *name = NULL;
+
+		vtable = *(void**)(object);
```

Comment:
Can we use `object->vtable` here instead?

Reply:
Or better, `GET_VTABLE`

Reply:
Can use `object->vtable` for sure. I think the awkward void cast was a holdover from stealing the `__try` trick from the debugger code.  This function isn't called from anywhere else. I've exposed it separately in the past because occasionally I like to call it directly from Unity land instead of leaning on the whole heap scanning. This is also why I didn't use the `GET_VTABLE` trick in this function as the MonoObject usually isn't marked.